File: lib/data/models/auth/auth_response_dto.dart
import 'package:json_annotation/json_annotation.dart';

part 'auth_response_dto.g.dart';

@JsonSerializable()
class AuthResponseDto {
  final String accessToken;
  final String refreshToken;
  final String userPublicId; // Java의 UUID는 String으로 매핑
  final String username;

  AuthResponseDto({
    required this.accessToken,
    required this.refreshToken,
    required this.userPublicId,
    required this.username,
  });

  // JSON 직렬화 로직
  factory AuthResponseDto.fromJson(Map<String, dynamic> json) =>
      _$AuthResponseDtoFromJson(json);

  Map<String, dynamic> toJson() => _$AuthResponseDtoToJson(this);
}

File: backend/src/main/java/com/cookstemma/cookstemma/dto/auth/AuthResponseDto.java
package com.cookstemma.cookstemma.dto.auth;

import com.cookstemma.cookstemma.domain.enums.Role;

import java.util.UUID;

public record AuthResponseDto(
        String accessToken,
        String refreshToken, // RTR로 인해 매번 새로 발급됨
        UUID userPublicId,
        String username,
        Role role
) {}